tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Framework.cs

```diff
@@ -36,13 +38,15 @@ internal static string GetUrlForSpan(this HttpRequestBase request, QueryStringMa
         /// <seealso cref="Configuration.ConfigurationKeys.FeatureFlags.BypassHttpRequestUrlCachingEnabled"/>
         internal static string GetUrlForSpan(this HttpRequest request, QueryStringManager queryStringManager, bool bypassHttpRequestUrl)
         {
-            if (bypassHttpRequestUrl)
+            var url = bypassHttpRequestUrl ? RequestDataHelper.BuildUrl(request) : RequestDataHelper.GetUrl(request);
+
+            if (url is not null)
             {
-                return HttpRequestUtils.GetUrl(RequestDataHelper.BuildUrl(request), queryStringManager);
+                return HttpRequestUtils.GetUrl(url, queryStringManager);
             }
             else
             {
-                return HttpRequestUtils.GetUrl(RequestDataHelper.GetUrl(request), queryStringManager);
+                return string.Empty;
```

Comment:
I wonder if this method should actually return `null` (and we should update the callsites to handle it)?

My reason being that the expectation is that this method returns a "valid" URL, but `string.Empty` isn't valid. WDYT?

Reply:
Sure. Taking a closer look, returning null would be more aligned with what we do with other similar tags. Updated. Thanks!